x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_hosts_url_state.ts

```diff
@@ -70,6 +71,7 @@ const reducer = (state: HostsState, action: Action): HostsState => {
 };
 
 export const useHostsUrlState = () => {
+  const fetch$ = useMemo(() => new Subject<'load'>(), []);
```

Comment:
This gives us more control over when we want to reload the data from snapshot API. Currently, the request is only made when the search state changes. But when the search state remains unchanged, the page wouldn't load the data. i.e:

- Unified search with absolute times (both `from` and `to`), but for some reason, there is no data. The user has the option to retry, but this was not working.

Reply:
One difference I see now is that after a change in the controls (selecting an operating system for example) the results are not filtered directly like before (after clicking refresh the results are filtered) I know that for the unified search there is an update button but we don't have it for the controls. So should we also reload when the user selects an option inside the controls
group? Also when the URL with a preselected OS is shared on the initial load the results are not filtered: 
https://user-images.githubusercontent.com/14139027/209811934-e536b6f8-8b9e-461d-a60f-f9a241247cad.mov

Reply:
Thanks for finding this problem. I had to rethink the whole solution to cover this scenario and also the double requests when the page is loaded with control filters applied.